File: GFG POTD/019-June-2025/011-remove-the-balls.java
// https://www.geeksforgeeks.org/problems/remove-the-balls--170647/1




import java.util.AbstractMap.SimpleEntry;
import java.util.Stack;

class Solution {
    public int findLength(int[] color, int[] radius) {
        int n = color.length;
        Stack<SimpleEntry<Integer, Integer>> stk = new Stack<>();
        for (int i = 0; i < n; i ++) {
            if (!stk.isEmpty() && stk.peek().getKey().equals(color[i]) && stk.peek().getValue().equals(radius[i])) {
                stk.pop();
            } else {
                stk.push(new SimpleEntry<>(color[i], radius[i]));
            }
        }
        return stk.size();
    }
}

File: GFG POTD/019-June-2025/011-remove-the-balls.cpp
// https://www.geeksforgeeks.org/problems/remove-the-balls--170647/1




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N)
    // SC : O(N)
    int findLength(vector<int> &color, vector<int> &radius) {
        int n = color.size();
        stack<pair<int, int>> stk;
        for (int i = 0; i < n; i ++) {
            if (!stk.empty() && stk.top().first == color[i] && stk.top().second == radius[i]) {
                stk.pop();
            } else {
                stk.push({color[i], radius[i]});
            }
        }
        return stk.size();
    }
};